crates/platform-tags/src/tags.rs

```diff
@@ -485,6 +487,10 @@ fn compatible_tags(platform: &Platform) -> Result<Vec<String>, PlatformError> {
             )));
         }
     };
+
+    // Tags are constructed in increasing order priority, reverse on return
+    platform_tags.reverse();
```

Comment:
I need to read up on it, but I don't think this is quite what we want. We need to change the ordering of the `manylinux2010_` and friends statements too.